scripts/portal.js

```diff
@@ -7,33 +7,43 @@ function Portal (sites) {
-  }
+  const _readme = '<p class="readme">This webring is an attempt to inspire artists & developers to build their own website and share traffic among each other. The ring welcomes personalized websites such as <b>diaries, wikis & portfolios</b>.</p><p>To add yourself to the ring, submit a <a href="https://github.com/XXIIVV/webring/edit/master/scripts/sites.js" target="_blank">Pull Request</a>.<br />If you found a broken link, please <a href="https://github.com/XXIIVV/webring/issues/new" target="_blank">report it</a>.</p>'
+  const _buttons = '<p class="buttons"><a href="#random" onClick="portal.reload()">Random</a> | <a href="https://github.com/XXIIVV/webring">Information</a> <a id="icon" href="#random" onClick="portal.reload()"></a> | <a href="hallway.html">The Hallway</a> | <a href="wiki.html">The Wiki</a> | <a id="opml">OPML</a></p>'
 
   function _directory (sites) {
     const siteTypesArray = [...new Set(sites.map(site => site.type).filter(Boolean))]
+    const siteLangsArray = [...new Set(sites
+      .map(site => site.langs)
+      .reduce((flattened, langs) => flattened.concat(langs), [])
+      .sort())]
+
     const siteTypes = siteTypesArray.reduce((output, siteType) =>
-      `${output}<a data-type="${siteType}" href="#" onclick="toggleType(event)">&lt;${siteType}&gt;</a>`,
-      '<a class="current" href="#" onclick="toggleType(event)">&lt;all&gt;</a>')
+      `${output}<a data-type="${siteType}" href="#" onclick="filterSites(event)">&lt;${siteType}&gt;</a>`,
+      '<a class="currentType" data-type="all" href="#" onclick="filterSites(event)">&lt;all&gt;</a>')
+
+    const siteLangs = siteLangsArray.reduce((output, siteLang) =>
+      `${output}<a data-lang="${siteLang}" href="#" onclick="filterSites(event)">&lt;${siteLang}&gt;</a>`,
+      '<a class="currentLang" data-lang="all" href="#" onclick="filterSites(event)">&lt;all&gt;</a>')
+
     const listItems = sites.reduce((acc, site, id) =>
-      `${acc}<li ${_type(site)} id='${id}'><a href='${site.url}'>${_name(site)}</a></li>`, '')
-    return `<nav>${siteTypes}</nav><main><ul>${listItems}</ul></main><footer>${_readme()}${_buttons()}</footer>`
+      `${acc}<li ${_type(site)} ${_lang(site)} id='${id}'><a href='${site.url}'>${_name(site)}</a></li>`, '')
+    return `<nav>${siteTypes}</nav><nav>${siteLangs}</nav><main><ul>${listItems}</ul></main><footer>${_readme}${_buttons}</footer>`
   }
 
   function _name (site) {
     return site.title || `${site.url.split('//')[1]}`
   }
 
   function _type (site) {
-    return site.type ? `data-type="${site.type}"` : ''
+    return site.type ? `data-type="${site.type} "` : ''
+  }
+
+  function _lang (site) {
+    const langs = site.langs.reduce((langs, lang) => `${langs} ${lang}`, '')
```

Comment:
I think that you can use, here, https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/join

```javascript
const langs = site.langs.join(' ')
```

Reply:
o duh, of course, way simpler.